pkg/backend/telemetryapi/request.go

```diff
@@ -33,17 +33,36 @@ type requestsBuilder interface {
 }
 
 var (
-	errUnableToSplit = fmt.Errorf("unable to split large payload further")
+	errUnableToSplit     = fmt.Errorf("unable to split large payload further")
+	maxEntitiesByRequest = 100
 )
 
 func newBatchRequest(ctx context.Context, metricsBatch []metricBatch, apiKey string, url string, userAgent string) (reqs []request, err error) {
-	// todo: split payload based on:
-	// a) number of entities being sent
-	// b) payload size
+	// todo: split payload based on payload size
 	if len(metricsBatch) < 1 {
 		return nil, nil
 	}
 
+	if len(metricsBatch) <= maxEntitiesByRequest {
+		return buildRequests(ctx, metricsBatch, apiKey, url, userAgent)
+	}
+
+	metrics := metricsBatch[:min(len(metricsBatch), maxEntitiesByRequest)]
+	req, err := buildRequests(ctx, metrics, apiKey, url, userAgent)
+	reqs = append(reqs, req...)
+
+	if len(metricsBatch[maxEntitiesByRequest:]) > 0 {
+		req, err = newBatchRequest(ctx, metricsBatch[maxEntitiesByRequest:], apiKey, url, userAgent)
```

Comment:
this creates a new batch with the left-overs of the previous, right?

therefore if limit is 10 and I get this calls:
- newBatchRequest(11)
- newBatchRequest(6)

the result will be 3 requests of sizes: 10, 1, 6

would it be possible to get two requests with sizes 11, 7 with a different approach?

Reply:
Mmmm interesting, let me work on that, thanks.